src/libraries/System.Private.CoreLib/src/System/Guid.cs

```diff
@@ -1048,121 +1155,421 @@ public bool TryFormat(Span<char> destination, out int charsWritten, ReadOnlySpan
+                Vector256<byte> last16CharsWithDashes = Avx2.Or(
+                    Avx2.Shuffle(rawLast16BytesChars, formatDLast16DashesShuffleMask),
+                    formatDLast16DashesSet);
+                // Will store: xxxxxxxxxxxxxxxxxx-8899-AABB______xx
+                Avx.Store((byte*)dest + 36, last16CharsWithDashes);
+                // Destination value after store: 00112233-4455-6677-8899-AABBxxxxxxxx
+
+                Vector256<byte> last16CharsRemainWithDashes = Avx2.Or(
+                    Avx2.Shuffle(rawLast16BytesChars, formatDLast16RemainDashesShuffleMask),
+                    formatDLast16RemainDashesSet);
+                // Will store: xxxxxxxxxxxxxxxxxxxx899-AABBCCDDEEFF
+                Avx.Store((byte*)dest + 40, last16CharsRemainWithDashes);
+                // Destination value after store: 00112233-4455-6677-8899-AABBCCDDEEFF
+            }
+        }
+
+        private unsafe void FormatNAvx(short* dest)
+        {
+            // dddddddddddddddddddddddddddddddd
+            fixed (Guid* thisPtr = &this)
+            {
+                Vector128<byte> layoutMask = s_layoutShiftMask;
+                Vector256<byte> asciiTable = s_asciiTable;
+                Vector128<byte> guidStringLayoutBytes = Ssse3.Shuffle(Sse2.LoadVector128((byte*)thisPtr), layoutMask);
+                Vector256<short> guidStringLayoutUtf16 = Avx2.ConvertToVector256Int16(guidStringLayoutBytes);
+                Vector256<byte> hi = Avx2.ShiftRightLogical(guidStringLayoutUtf16, 4).AsByte();
+                Vector256<byte> lo = Avx2.ShiftLeftLogical(guidStringLayoutUtf16, 8).AsByte();
+                Vector256<byte> asciiTableIndices = Avx2.And(Avx2.Or(hi, lo), Vector256.Create((byte)0x0F));
+                Vector256<byte> asciiBytes = Avx2.Shuffle(asciiTable, asciiTableIndices);
+                Avx.Store(dest, Avx2.ConvertToVector256Int16(asciiBytes.GetLower()));
+                Avx.Store(dest + 16, Avx2.ConvertToVector256Int16(asciiBytes.GetUpper()));
+            }
+        }
+
+        private unsafe void FormatD(char* dest)
+        {
+            // dddddddd-dddd-dddd-dddd-dddddddddddd
+            uint* destUints = (uint*)dest;
+            char** destUintsAsChars = (char**)&destUints;
+            uint* tableToHex = TableToHex;
+
+            dest[8] = dest[13] = dest[18] = dest[23] = '-';
+            destUints[0] = tableToHex[(byte)(_a >> 24)];
+            destUints[1] = tableToHex[(byte)(_a >> 16)];
+            destUints[2] = tableToHex[(byte)(_a >> 8)];
+            destUints[3] = tableToHex[(byte)_a];
+            destUints[7] = tableToHex[(byte)(_c >> 8)];
+            destUints[8] = tableToHex[(byte)_c];
+            destUints[12] = tableToHex[_f];
+            destUints[13] = tableToHex[_g];
+            destUints[14] = tableToHex[_h];
+            destUints[15] = tableToHex[_i];
+            destUints[16] = tableToHex[_j];
+            destUints[17] = tableToHex[_k];
+            *destUintsAsChars += 1;
```

Comment:
I don't feel comfortable using pointers in this manner to perform unaligned memory accesses.

Reply:
But that offsets helps to understand [how it works](https://gist.githubusercontent.com/vanbukin/490180dc1a76f4407b0d8f4f1f00d9c4/raw/25166534b41ac581dbd1177e5932fd41adaba102/offsets.png). It may be difficult to support that code without them. Especially for the "X" format. Should I remove them?